File: Backend/src/EduCrm.Domain/Entities/Schedule.cs
namespace EduCrm.Domain.Entities;

public class Schedule
{
    public int Id { get; set; }

    public int GroupId { get; set; }
    public DayOfWeek DayOfWeek { get; set; }
    public TimeSpan StartTime { get; set; }
    public TimeSpan EndTime { get; set; }
    public string? Room { get; set; }

    public DateTime RecurringFrom { get; set; }
    public DateTime? RecurringTo { get; set; }

    public Group Group { get; set; } = null!;
}

File: Frontend/web/src/types/schedule.ts
export interface ScheduleResponse {
    id: number;
    groupId: number;
    groupName: string;
    dayOfWeek: string;        // "Monday", "Tuesday"...
    startTime: string;        // "10:00:00"
    endTime: string;          // "12:00:00"
    room: string | null;
    recurringFrom: string;    // ISO дата
    recurringTo: string | null;
}

export interface CreateScheduleRequest {
    groupId: number;
    dayOfWeek: number;     
    startTime: string;
    endTime: string;
    room?: string;
    recurringFrom: string;
    recurringTo?: string;
}

export interface UpdateScheduleRequest {
    dayOfWeek: number;      
    startTime: string;
    endTime: string;
    room?: string;
    recurringTo?: string;
}